wandb/sdk/integration_utils/data_logging.py

```diff
@@ -274,16 +274,21 @@ def _infer_single_example_keyed_processor(
         )
         # Assume these are logits
         class_names = class_labels_table.get_column("label")
+
         processors["max_class"] = lambda n, d, p: class_labels_table.index_ref(  # type: ignore
             np.argmax(d)
         )
         # TODO: Consider adding back if users ask
         # processors["min_class"] = lambda n, d, p: class_labels_table.index_ref(  # type: ignore
         #     np.argmin(d)
         # )
-        processors["score"] = lambda n, d, p: {
-            class_names[i]: d[i] for i in range(shape[0])
-        }
+
+        values = np.unique(example)
+        is_one_hot = len(values) == 2 and set(values) == set([0, 1])
```

Comment:
What happens if we detect this in error? We just end up with some extra columns?

For example, any vector of length 1 may look like a 1-hot vector, like you could have [1, 0], but also [.707, .707]

Reply:
Ultimately, this change returns strictly less columns than before. Currently we have extra columns in the one_hot cases. Maybe I am missing your point here?

I think `[.707, .707]` would return false for is_one_hot.

Reply:
I don't remember how this code is structured. I think we're looking at cells within a column, one at a time, and deciding what to do. Since we look at one example at a time, (at least, that's what I'm assuming), we can't tell if a column is actually 1-hot encoding, or is a vector of length 1, by looking a a single example.